### DISCUSSION TITLE
App authentication Error

### DISCUSSION THREAD

**Discussion by Aliza279** [kinngh/shopify-node-mongodb-next-app#48] [General]
Thank you for creating a starter kit, Can anyone please let me know by using this repo if I create a Custom distribution link?
It works fine at the start but after some time I have to use `{{BASE_URL}}/api/auth?shop={{shopname}}.myshopify.com` to update auth token in the DB because old tokens expire after some time. Sometimes I also notice that the custom distribution link doesn't work directly and I have to use that specific link I mentioned above again.
How can I avoid this?

**Comment by Aliza279** [comment]
@kinngh Can you please help me out if possible?

**Reply by kinngh** [reply]
I haven't had any issues with custom distribution links - though if you're worried about online tokens just swap out the verify request to use offline session tokens and swap out the verification strait and you'll be good to go!
